src/chronos/chronos.py

```diff
@@ -306,6 +336,51 @@ def __init__(self, tokenizer, model):
         self.tokenizer = tokenizer
         self.model = model
 
+    def _prepare_and_validate_context(
+        self, context: Union[torch.Tensor, List[torch.Tensor]]
+    ):
+        if isinstance(context, list):
+            context = left_pad_and_stack_1D(context)
+        assert isinstance(context, torch.Tensor)
+        if context.ndim == 1:
+            context = context.unsqueeze(0)
+        assert context.ndim == 2
+
+        return context
+
+    @torch.no_grad()
+    def embed(
+        self, context: Union[torch.Tensor, List[torch.Tensor]]
+    ) -> Tuple[torch.Tensor, torch.Tensor]:
+        """
+        Get encoder embeddings for the given time series.
+
+        Parameters
+        ----------
+        context
+            Input series. This is either a 1D tensor, or a list
+            of 1D tensors, or a 2D tensor whose first dimension
+            is batch. In the latter case, use left-padding with
+            ``torch.nan`` to align series of different lengths.
+
+        Returns
+        -------
+        embeddings, scale
+            A tuple of two tensors: the encoder embeddings and the time series scale.
+            The encoder embeddings are shaped (batch_size, context_length, d_model)
+            or (batch_size, context_length + 1, d_model), where the extra 1 is for EOS.
+            If your original time series were shorter than the model's context_length,
+            please slice the returned embeddings along the time axis accordingly.
```

Comment:
Maybe this sentence can be remove if #25 is merged

Reply:
Yes.

Reply:
Done.